### PR TITLE
Extracted JVM to dependencies.cmake

### PR DESCRIPTION
**Pull request by l4l**
There's module FindJNI in cmake that works pretty awful, anyway it support more wide amount of platforms
That's module doesn't handle `JAVA_HOME` so checks it in advance

### CODE REVIEW

**Comment by l4l** [comment]
there's still issues, please don't merge

File: core/util/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,6 @@
 set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
 set(JSON_PATH "${PROJECT_SOURCE_DIR}/core/vendor/json")
```
Comment (Warchant) [inline core/util/CMakeLists.txt]: remove

File: core/validation/CMakeLists.txt [hunk 2]
```diff
@@ -5,13 +5,11 @@ SET(ED25519_PATH "${PROJECT_SOURCE_DIR}/core/vendor/ed25519")
 
 include_directories(
   ${ED25519_PATH}/src
```
Comment (Warchant) [inline core/validation/CMakeLists.txt]: remove

File: core/validation/CMakeLists.txt [hunk 3]
```diff
@@ -5,13 +5,11 @@ SET(ED25519_PATH "${PROJECT_SOURCE_DIR}/core/vendor/ed25519")
...
-  ${PROJECT_SOURCE_DIR}/core
 )
 
 link_directories(
```
Comment (Warchant) [inline core/validation/CMakeLists.txt]: remove

File: dependencies.cmake [hunk 4]
```diff
@@ -15,12 +15,12 @@ find_package(LibXslt QUIET)
...
   message(FATAL_ERROR "xsltproc not found")
 endif ()
-set(MAKE_C_FLAGS "-fpermissive")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DKeccakP200_excluded -DKeccakP400_excluded -DKeccakP800_excluded ")
+
 ExternalProject_Add(gvanas_keccak
   GIT_REPOSITORY    "https://github.com/gvanas/KeccakCodePackage.git"
   BUILD_IN_SOURCE   1
-  BUILD_COMMAND     bash -c "CFLAGS='-fPIC -fpermissive' $(MAKE) CC='${CMAKE_CXX_COMPILER}' generic64/libkeccak.a"
+  BUILD_COMMAND     bash -c "CFLAGS='-fPIC -fpermissive -DKeccakP200_excluded -DKeccakP400_excluded -DKeccakP800_excluded'\
```
Comment (Warchant) [inline dependencies.cmake]: remove `-fpermissive`

File: peer/CMakeLists.txt [hunk 5]
```diff
@@ -1,13 +1,6 @@
...
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
 
-
-SET(CMAKE_CXX_FLAGS "-g -std=c++1y -Wall -fPIC")
-
 include_directories(
-  ${PROJECT_SOURCE_DIR}/core
-  ${PROJECT_SOURCE_DIR}/core/server
-  ${PROJECT_SOURCE_DIR}/core/peer
   ${LEVELDB_PATH}/include
```
Comment (Warchant) [inline peer/CMakeLists.txt]: remove

File: smart_contract/repository/CMakeLists.txt [hunk 6]
```diff
@@ -1,20 +1,13 @@
...
 set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
-set(CMAKE_CXX_FLAGS "-g -std=c++1y -Wall -fPIC")
 
 include_directories(
-  /usr/lib/jvm/java-1.8.0-openjdk-amd64/include
-  /usr/lib/jvm/java-1.8.0-openjdk-amd64/include/linux
-  ${JAVA_HOME}/include
-  ${JAVA_HOME}/include/linux
-  ${PROJECT_SOURCE_DIR}/core
   ${LEVELDB_PATH}/include
```
Comment (Warchant) [inline smart_contract/repository/CMakeLists.txt]: remove

File: smart_contract/repository/CMakeLists.txt [hunk 7]
```diff
@@ -1,20 +1,13 @@
...
 )
 
 link_directories(
```
Comment (Warchant) [inline smart_contract/repository/CMakeLists.txt]: remove

change to 
```
target_link_libraries(TARGET123
leveldb
)
```

File: test/consensus/CMakeLists.txt [hunk 8]
```diff
@@ -1,13 +1,7 @@
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/my_test_bin)
 
-SET(CMAKE_CXX_FLAGS_DEBUG "-g")
-
 SET(LEVELDB_PATH "${PROJECT_SOURCE_DIR}/core/vendor/leveldb")
```
Comment (Warchant) [inline test/consensus/CMakeLists.txt]: remove

File: test/infra/service/CMakeLists.txt [hunk 9]
```diff
@@ -3,10 +3,6 @@ set(CMAKE_BUILD_TYPE Debug)
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/test_bin)
 set(JSON_PATH "${PROJECT_SOURCE_DIR}/core/vendor/json")
```
Comment (Warchant) [inline test/infra/service/CMakeLists.txt]: remove

File: core/validation/CMakeLists.txt [hunk 10]
```diff
@@ -3,10 +3,6 @@ SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
 SET(KECCAK_PATH  "${PROJECT_SOURCE_DIR}/core/vendor/KeccakCodePackage")
 SET(ED25519_PATH "${PROJECT_SOURCE_DIR}/core/vendor/ed25519")
 
-include_directories(
-  ${ED25519_PATH}/src
-)
-
 link_directories(
```
Comment (Warchant) [inline core/validation/CMakeLists.txt]: remove

instead link 
`leveldb`
`keccak`

File: peer/CMakeLists.txt [hunk 11]
```diff
@@ -1,9 +1,5 @@
 SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
 
-include_directories(
-  ${LEVELDB_PATH}/include
-)
-
 link_directories(
```
Comment (Warchant) [inline peer/CMakeLists.txt]: remove

change to 
```
taget_link_libraries (target
leveldb
... (all necessary)
```

File: smart_contract/repository/CMakeLists.txt [hunk 12]
```diff
@@ -2,10 +2,6 @@ cmake_minimum_required(VERSION 3.3.0 FATAL_ERROR)
 
 set(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
 
-include_directories(
-  ${LEVELDB_PATH}/include
-)
-
 link_directories(
```
Comment (Warchant) [inline smart_contract/repository/CMakeLists.txt]: remove

change to 
```
taget_link_libraries (target
leveldb
... (all necessary)
```

File: smart_contract/repository/CMakeLists.txt [hunk 13]
```diff
@@ -17,11 +13,7 @@ add_library(Repository SHARED
...
 
 target_link_libraries(Repository
-  leveldb
   core_repository
-  virtual_machine_with_java
-  event_with_grpc
-  create_objects_helper
 )
 
 add_custom_command(TARGET Repository
```
Comment (Warchant) [inline smart_contract/repository/CMakeLists.txt]: what is this?

File: test/infra/repository/CMakeLists.txt [hunk 14]
```diff
@@ -1,7 +1,3 @@
-SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/test_bin)
-
-set(CMAKE_BUILD_TYPE Debug)
-
 link_directories(
```
Comment (Warchant) [inline test/infra/repository/CMakeLists.txt]: remove

change to 
```
taget_link_libraries (target
... (all necessary)
```

File: test/vendor/thread_pool_test.cpp [hunk 15]
```diff
@@ -282,10 +282,10 @@ TEST(ThreadPool, Sequential_WithStdQueue) {
   const int repetitions = 100;
 
   for (int i = 1; i <= repetitions; i++) {
-    queue.push(std::move(pool.process([i]() {
+    queue.push(pool.process([i]() {
```
Comment (Warchant) [inline test/vendor/thread_pool_test.cpp]: pool.process is lvalue here, std::move is necessary to not to make copies

Comment (l4l) [inline test/vendor/thread_pool_test.cpp]: Ah, that fix was by accident actually

Comment (l4l) [inline test/vendor/thread_pool_test.cpp]: and btw clang doesn't agree with you

**Pull request merged**